pkg/auth/oauth/registry/registry_test.go

```diff
@@ -473,3 +476,204 @@ func TestAuthenticateTokenValidated(t *testing.T) {
 		t.Error("Did not get a user!")
 	}
 }
+
+type fakeOAuthClientLister struct {
+	clients oauthclient.OAuthClientInterface
+}
+
+func (f fakeOAuthClientLister) Get(name string) (*oapi.OAuthClient, error) {
+	return f.clients.Get(name, metav1.GetOptions{})
+}
+
+func (f fakeOAuthClientLister) List(selector labels.Selector) ([]*oapi.OAuthClient, error) {
+	var list []*oapi.OAuthClient
+	ret, _ := f.clients.List(metav1.ListOptions{})
+	for i := range ret.Items {
+		list = append(list, &ret.Items[i])
+	}
+	return list, nil
+}
+
+func checkToken(t *testing.T, name string, authf authenticator.Token, tokens oauthclient.OAuthAccessTokenInterface, present bool) {
+	userInfo, found, err := authf.AuthenticateToken(name)
+	if present {
+		if !found {
+			t.Errorf("Did not find token %s!", name)
+		}
+		if err != nil {
+			t.Errorf("Unexpected error checking for token %s: %v", name, err)
+		}
+		if userInfo == nil {
+			t.Errorf("Did not get a user for token %s!", name)
+		}
+	} else {
+		if found {
+			token, _ := tokens.Get(name, metav1.GetOptions{})
+			t.Errorf("Found token (created=%s, timeout=%di, now=%s), but it should be gone!",
+				token.CreationTimestamp, token.InactivityTimeoutSeconds, time.Now())
+		}
+		if err != errTimedout {
+			t.Errorf("Unexpected error checking absence of token %s: %v", name, err)
+		}
+		if userInfo != nil {
+			t.Errorf("Unexpected user checking absence of token %s: %v", name, userInfo)
+		}
+	}
+}
+
+func TestAuthenticateTokenTimeout(t *testing.T) {
```

Comment:
I think we will want a test for retry logic, and another test that "counts" reads/writes to ETCD (will probably have to be fuzzy).

Reply:
add followup issue if you think we really need it, I am not sure we do